test/acsengine-features.groovy

```diff
@@ -37,6 +37,9 @@ node {
                 env.SUBSCRIPTION_ID="${SUBSCRIPTION_ID}"
                 env.CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID="${CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID}"
                 env.CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET="${CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET}"
+                if(CUSTOM_HYPERKUBE_SPEC) {
```

Comment:
have you tried this on a Jenkins job without this parameter? IIRC, Jenkins loves to throw undefined property exception of sort.....